test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -243,26 +241,10 @@ public Task RemoteIPv6Address()
                     {
                         await context.Request.Body.ReadAsync(new byte[1], 0, 1);
                     }
-                    catch (BadHttpRequestException)
-                    {
-                        // We need this here because BadHttpRequestException derives from IOException,
-                        // and we're looking for an actual IOException.
-                    }
                     catch (IOException ex)
                     {
-                        // This is one of two exception types that might thrown in this scenario.
-                        // An IOException is thrown if ReadAsync is awaiting on SocketInput when
-                        // the connection is aborted.
                         expectedExceptionThrown = ex.InnerException is UvException;
```

Comment:
Checking `StatusCode` is not optimal because 1) we lose the advantage of not needing platform specific code anymore to determine the correct value of `ECONNRESET` and 2) I'd have to expose internals to the functional tests project.

I'm going to check for `"ECONNRESET"` in the exception message like I did in the other test.